python/lgbserver/lgbserver/model.py

```diff
@@ -42,16 +42,48 @@ def load(self) -> bool:
             raise ModelMissingError(model_file)
         self._booster = lgb.Booster(params={"nthread": self.nthread},
                                     model_file=model_file)
+        self._categorical_columns = self._load_categorical(self._booster, model_file)
         self.ready = True
         return self.ready
 
+    def _load_categorical(
+        self, booster: Booster, model_file: str
+    ) -> Optional[List[str]]:
+        # LightGBM does not currently force trained categorical columns to
+        # categorical during predict, so pull the `categorical_feature` from the
+        # saved model
+        # https://github.com/microsoft/LightGBM/issues/5244
```

Comment:
there is another implementation, not sure if it is better
https://github.com/microsoft/LightGBM/issues/5244#issuecomment-1270739681